### PR TITLE
[Backport][ipa-4-6] Test to check if Certmonger tracks certs in between reboots/interruptions

### PR DESCRIPTION
**Pull request by rcritten**
This PR was opened as a manual cherry-pick of PR #4373 which was pushed to master and backport to ipa-4-6 is required.

Rebased ipatests/pytest_ipa/integration/tasks.py

### CODE REVIEW

**Comment by rcritten** [comment]
Just need a quick once-over. The backport wasn't difficult but another set of eyes is a good idea.

File: ipatests/test_integration/test_cert.py [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+
+    def test_cacert_file_appear_with_option_F(self):
+        """Test if getcert creates cacert file with -F option
+
+        It took longer to create the cacert file in older version.
+        restarting the certmonger service creates the file at the location
+        specified by -F option. This fix is to check that cacert file
+        creates immediately after certificate goes into MONITORING state.
+
+        related: https://pagure.io/freeipa/issue/8105
```
Comment (mrizwan93) [inline ipatests/test_integration/test_cert.py:78]: Fix for this issue is not available in ipa-4-6. Hence this test not required here.

Comment (rcritten) [reply ipatests/test_integration/test_cert.py:78]: Yeah, that whole class was added somehow. Removing, good catch.

**Comment by rcritten** [comment]
Ok fixed, thanks. I don't know how I let this go so sideways...

**Comment by flo-renaud** [comment]
Hi @rcritten 
Sorry for noticing this so late, but since test_cert.py is a new file, it needs to be added to the nightly and gating definitions.

**Comment by flo-renaud** [comment]
Hi @rcritten 
Tests are green and the backport LGTM. Please remove the temp commit and I will Ack.

File: ipatests/prci_definitions/gating.yaml [hunk 2]
```diff
@@ -98,3 +98,15 @@ jobs:
...
+
+  fedora-27/test_cert:
+    requires: [fedora-27/build]
+    priority: 100
+    job:
+      class: RunPytest
+      args:
+        build_url: '{fedora-27/build_url}'
+        test_suite: test_integration/test_cert.py
+        template: *ci-master-27
```
Comment (netoarmando) [inline ipatests/prci_definitions/gating.yaml:110]: Missing letter `f`, it should be: `*ci-master-f27`.

File: ipatests/prci_definitions/nightly_ipa-4-6.yaml [hunk 3]
```diff
@@ -1134,3 +1134,15 @@ jobs:
...
+
+  fedora-27/test_cert:
+    requires: [fedora-27/build]
+    priority: 50
+    job:
+      class: RunPytest
+      args:
+        build_url: '{fedora-27/build_url}'
+        test_suite: test_integration/test_cert.py
+        template: *ci-master-27
```
Comment (netoarmando) [inline ipatests/prci_definitions/nightly_ipa-4-6.yaml:1146]: Missing letter `f`, it should be: `*ci-master-f27`

Comment (rcritten) [reply ipatests/prci_definitions/nightly_ipa-4-6.yaml]: Fixed. I made the same mistake in gating.yaml

**Comment by rcritten** [comment]
ipa-4-6:

* f77c2f122802aad38aa12e5fc6c9cc49aaa49dfa Test to check if Certmonger tracks certs in between reboots/interruptions and while in "CA_WORKING" state
* 4235ccba769653fc0d906d4c79bdde5dbfcdad74 ipatests: run test_integration/test_cert.py in PR-CI

**Pull request closed without merging**